src/fides/api/util/tcf/experience_meta.py

```diff
@@ -46,18 +63,74 @@ def _build_tcf_version_hash_model(
     get the maximum possible number of attributes added to our hash for the current
     system configuration.
     """
+
+    def build_vendor_by_purpose_strings(
+        vendor_list: Union[
+            List[TCFVendorConsentRecord], List[TCFVendorLegitimateInterestsRecord]
+        ]
+    ) -> List[str]:
+        """Hash helper for building vendor x purposes.  Converts a list of vendor records
+        into a list of strings each consisting of a vendor id, a separator (-), and comma
+        separated purposes (if applicable) with the given legitimate interest.
+
+        Example:
+        [<universal_vendor_id>-<comma separated purposes>]
+        ["gacp.100", "gvl.8-1,2,3", "gvl.56-3,4,5]
+        """
+        vendor_by_purpose_lists: List[str] = []
+        for vendor_record in vendor_list:
+            vendor_id: str = vendor_record.id
+            purposes_key: str = (
+                "purpose_consents"
+                if isinstance(vendor_record, TCFVendorConsentRecord)
+                else "purpose_legitimate_interests"
+            )
+            purpose_ids = sorted(
+                purpose_record.id
+                for purpose_record in getattr(vendor_record, purposes_key)
+            )
```

Comment:
Purposes should already be in order from the base experience but just in case, they are sorted here again -